scripts/globals/znm.lua

```diff
@@ -46,9 +118,33 @@ xi.znm.soultrapper.onItemCheck = function(target, user)
     return 0
 end
 
+-----------------------------------
+-- onItemUse
+-----------------------------------
+xi.znm.soultrapper.onItemUse = function(target, user, item)
+    -- Soul plate not guaranteed
+    if math.random(100) > xi.znm.SOULTRAPPER_SUCCESS then
+        user:messageBasic(xi.msg.basic.SOULTRAPPER_FAILED)
+    else
+        -- Determine Zeni starting value
+        local zeni = xi.znm.soultrapper.getZeniValue(target, user, item)
+
+        -- Pick a skill totally at random...
+        local skillIndex, skillEntry = utils.randomEntryIdx(xi.pankration.feralSkills)
+
+        -- Add plate
+        local plate = user:addSoulPlate(target:getName(), target:getFamily(), zeni, skillIndex, skillEntry.fp)
+        local data = plate:getSoulPlateData()
+        utils.unused(data)
+        user:messageBasic(xi.msg.basic.SOULTRAPPER_SUCCESS, 0, xi.items.SOUL_PLATE)
+    end
+end
+
+-----------------------------------
+-- onItemUse Helpers
+-----------------------------------
 xi.znm.soultrapper.getZeniValue = function(target, user, item)
```

Comment:
At least from 75 era sources, there was a lot of conjecture and partial proof of two more components impacting zeni value,

Mob Size - larger size, more zeni.  Additionally there were therories that the zeni bonuses branched based on mob size.
Instances - Things like Nyzul Isle and Salvage were reported to give better zeni for the same level mobs found outside.  This is why wings has a "beingIrresponsible" zeni modifier that awards a bonus for taking picutues during instanced combat  -when the player should be focused on their objective.

Again, community data is unverified and should be captured

Some sources for Zeni per mob in the 75 era can be found here, as well on differing wiki talk pages
https://www.bluegartr.com/threads/63949-Zeni-per-Image-rates  (has some theories on size based value)
http://ffxiaydel.blogspot.com/2009/03/zeni-point-farming.html
https://wikiwiki.jp/ffxi/NM/%E8%8A%B1%E9%B3%A5%E9%A2%A8%E6%9C%88 (has a table of what they believe is part of zeni calc which includes size)
http://wiki.ffo.jp/html/1155.html ( has more detailed laid out points but did not often fit real world examples like flower nm)